primer/brand
# SubdomainNavBar component

## Summary

![nav bar in dark mode with overflow menu open](https://user-images.githubusercontent.com/13340707/191292289-dac3021a-de94-44e5-8824-05e1d0245d3a.png)


Adds a baseline `SubdomainNavBar` component, which fulfils the design and semantic markup requirements for this component. Future iterations will focus on increasing test coverage and improving aspects of accessibility. 

API example:
```jsx
<SubdomainNavBar title="Subdomain">
  <SubdomainNavBar.Link href="#">Collections</SubdomainNavBar.Link>
  <SubdomainNavBar.Link href="#">Topics</SubdomainNavBar.Link>
  <SubdomainNavBar.Link href="#">Articles</SubdomainNavBar.Link>
  <SubdomainNavBar.Link href="#">Events</SubdomainNavBar.Link>
  <SubdomainNavBar.Link href="#">Video</SubdomainNavBar.Link>
  <SubdomainNavBar.Link href="#">Social</SubdomainNavBar.Link>
  <SubdomainNavBar.Search
    ref={inputRef}
    onSubmit={handleSubmit}
  />
  <SubdomainNavBar.PrimaryAction href="#">Primary CTA</SubdomainNavBar.PrimaryAction>
  <SubdomainNavBar.SecondaryAction href="#">Secondary CTA</SubdomainNavBar.SecondaryAction>
</SubdomainNavBar>
```


## What should reviewers focus on?

- Try it out in Storybook
  - [Light mode](https://primer-b5355bd8fd-26139705.drafts.github.io/storybook/?path=/story/components-subdomainnavbar--playground)
  - [Dark mode](https://primer-b5355bd8fd-26139705.drafts.github.io/storybook/?path=/story/components-subdomainnavbar--playground&globals=colorMode:dark)
- Review the API
- Check the React implementation
- Provide any feedback or missing functionality that you've noticed

## Steps to test:



1. Open links above
1. Interact with the menu and ensure behaviour is correct across all viewports
1. Verify design is per the specification

## Supporting resources (related issues, external links, etc):

- [Design file and specifications](https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/Primer-Brand?node-id=1024%3A32795)

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

> Please try to provide before and after screenshots or videos

https://user-images.githubusercontent.com/13340707/191295762-500de6fe-cbe1-4263-bc45-384a9abd916a.mov
